Stas

stanley-dds opened the pull request.

stanley-dds committed:
+gameBoard(raw)

stanley-dds committed:
+GameBoard(raw)

stanley-dds committed:
++fields,finals,nests;
an obj of <Game> class contains that data;
the gameboard ready for further developing;
the obj packed inside of <GameBoardViewModel> class

emilojonsson commented on Views/MainMenu.xaml.cs:
```diff
@@ -27,54 +27,12 @@ public MainMenu()
-                args.DrawingSession.DrawImage(renderTarget);
-            }
-        }
+            Frame.Navigate(typeof(GameBoard));
```
MainFrame.Navigate(typeof(StartMenu));

it should navigate to StartMenu instead

stanley-dds replied:
sure

emilojonsson commented on Views/GameBoard.xaml:
```diff
@@ -4,18 +4,24 @@
     Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
 
     <Grid>
         <TextBox 
```
Please remove this old testing code. Same thing below regarding TextBlock :)

emilojonsson commented on Views/GameBoard.xaml:
```diff
@@ -4,18 +4,24 @@
+        
+
+        <xaml:CanvasControl x:Name="GameBoardCanvas" Width="1000" Height="800" />
+        <TextBlock HorizontalAlignment="Left" Name="counter" Margin="42,649,0,0" TextWrapping="Wrap" Text="TextBlock" VerticalAlignment="Top"/>
```
This testing code could also be removed

emilojonsson commented on ViewModels/GameBoardViewModel.cs:
```diff
@@ -1,16 +1,27 @@
 {
     internal class GameBoardViewModel : ViewModelBase
     {
         private string _textStorlek;
```
Please remove old testing stuff

emilojonsson commented on ViewModels/GameBoardViewModel.cs:
```diff
@@ -1,16 +1,27 @@
+
+        public static GameBoardViewModel Instance { get; set; }
+
         public string TextStorlek {  
```
Same here, remove old testing stuff

emilojonsson commented on ViewModels/GameBoardViewModel.cs:
```diff
@@ -21,10 +32,253 @@ internal class GameBoardViewModel : ViewModelBase
+
         public GameBoardViewModel( ) 
         {
             this._textStorlek = "25";
```
same here, old testing stuff to be removed

emilojonsson commented on ViewModels/GameBoardViewModel.cs:
```diff
@@ -21,10 +32,253 @@ internal class GameBoardViewModel : ViewModelBase
         public GameBoardViewModel( ) 
         {
             this._textStorlek = "25";
             Location nest = new Location(new List<Piece>(), Helpers.TypeOfLocation.Nest);
```
same here, this is testing of the Location class...

stanley-dds replied:
sure;
Cz it seemed to be the beginning of locations that require to navigate with a piece. Probably not :)

emilojonsson commented on Models/Board.cs:
```diff
@@ -8,5 +8,9 @@ namespace LUDO.Models
 {
     internal class Board
     {
+        public List<Cell> GameCells { get; set; }
```
Are we working on a single List solution instead of several arrays that we initially discussed (see umlet)? Just as long as it is simpler and it works :)

emilojonsson reviewed (changes requested):
The Drawboard method looks nice and if I understand it we can later visualize pieces via the cells coordinates. Good work!

Please take a look at the comments regarding "Cells" or "Location", and "single List-" or "array"-solution regarding how we organize the cells/locations in the board.

I also made some notes regarding old test code that we just could delete.

stanley-dds committed:
-- <Location> is now <Cell>
++ fields,finals, finish and nestCells are all objs of <Cell> class

++to navigate a piece, use cell´s ID and boolean isFinal:
field: ID1 to 13 (isFinal false)
finish: ID 1 to 5 (isFinal true)
final: ID 6 (isFinal true)
nestCell: ID -1 to -4 (isFinal false)

emilojonsson reviewed (approved):
Looks good to me

emilojonsson committed:
Merge branch 'master' into Stas
